### DISCUSSION TITLE
mutual TLS &  username from certificate

### DISCUSSION THREAD

**Discussion by mntnclmbr** [halfgaar/FlashMQ#27] [Q&A]
Hi, 
I had a couple of questions on how tls was setup:
1. Looks like "mutual" TLS is required for TLS and both broker + client certs need to be signed using the same CA. Is that correct?
2. is/can the username be pulled out of the TLS client certificate? If this is how it is set up, which field is used for this and is it customizable?

**Comment by halfgaar** [comment]
The TLS is just on the listener (server identify check and transport encryption). There is no client authentication using certificates.

**Comment by mntnclmbr** [comment]
Understood. Is certificate based client authentication on the roadmap?

**Reply by halfgaar** [reply]
Not short term. But, I could look into it. I would like to hear more about your use case then.

**Reply by mntnclmbr** [reply]
That's ok. it is a good to have feature for our use case for now where we authenticate IoT devices with the TLS cert.

**Reply by itweb62** [reply]
Hello ,
Mtls is on the roadmap in short term , do you have a approx date for this feature ? 

Thank you

**Reply by halfgaar** [reply]
I can't really give a date, but I did make an issue for it: https://github.com/halfgaar/FlashMQ/issues/45

That way, it'll be on the short-list.

**Reply by halfgaar** [reply]
See ticket update. Feature is coming soon.

**Comment by halfgaar** [comment]
Closing discussion, because we have #45. Subscribe to that issue for updates.
